legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabRevisionApi.java

```diff
@@ -76,14 +77,15 @@ public RevisionAccessContext getProjectEntityRevisionContext(String projectId, S
             throw new LegendSDLCServerException("Invalid entity path: " + entityPath, Status.BAD_REQUEST);
         }
         ProjectFileAccessProvider fileAccessProvider = getProjectFileAccessProvider();
-        ProjectFileAccessProvider.FileAccessContext fileAccessContext = fileAccessProvider.getFileAccessContext(projectId, null, ProjectFileAccessProvider.WorkspaceAccessType.WORKSPACE, null);
+        ProjectFileAccessProvider.FileAccessContext fileAccessContext = fileAccessProvider.getFileAccessContext(projectId, null, null, null);
         ProjectStructure projectStructure = ProjectStructure.getProjectStructure(fileAccessContext);
         String filePath = projectStructure.findEntityFile(entityPath, fileAccessContext);
-        if (filePath == null)
+        String canonicalizedFilePath = ProjectPaths.canonicalizeFile(filePath);
+        if (canonicalizedFilePath == null)
```

Comment:
This check should be done on filePath and prior to canonicalizing. The call to findEntityFile is what actually checks that the entity file exists.